cpp/src/io/orc/stats_enc.cu

```diff
@@ -0,0 +1,380 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "orc_common.h"
+#include "orc_gpu.h"
+#include <io/utilities/block_utils.cuh>
+
+namespace cudf {
+namespace io {
+namespace orc {
+namespace gpu {
+
+/**
+ * @brief Initializes statistics groups
+ *
+ * @param[out] groups Statistics groups
+ * @param[in] cols Column descriptors
+ * @param[in] num_columns Number of columns
+ * @param[in] num_rowgroups Number of rowgroups
+ * @param[in] row_index_stride Rowgroup size in rows
+ *
+ **/
+// blockDim {128,1,1}
+__global__ void __launch_bounds__(128)
+gpuInitStatisticsGroups(statistics_group *groups, const stats_column_desc *cols,
+                        uint32_t num_columns, uint32_t num_rowgroups, uint32_t row_index_stride) {
+  __shared__ __align__(4) volatile statistics_group grp_g[4];
```

Comment:
What does `grp_g` mean? What is the _g for? Please avoid unnecessary abbreviation.

Reply:
A hint that is is shared (global var vs local definition). Convention inherited from other codebases.